castervoice/lib/ccr/core/text_manipulation.py

```diff
@@ -0,0 +1,254 @@
+from dragonfly import Function, Key, Text, Mouse, Pause, Dictation, Choice, Grammar , ContextAction
+
+
+from castervoice.lib import control, settings, text_manipulation_functions
+
+from castervoice.lib.context import AppContext
+from castervoice.lib.dfplus.additions import IntegerRefST
+from castervoice.lib.dfplus.merge import gfilter
+from castervoice.lib.dfplus.merge.mergerule import MergeRule
+from castervoice.lib.dfplus.merge.ccrmerger import CCRMerger
+from castervoice.lib.dfplus.state.short import R
+from castervoice.apps import utils
+from castervoice.apps import reloader
+from castervoice.apps import reloader
+
+
+
+class TextManipulation(MergeRule):
+    pronunciation = "text manipulation"
+
+
+    mapping = {
+
+        
+        # requires the latest version of dragonfly because of her recent modification of the Function action
+            # I think dragonfly2-0.13.0
+        # the wait times in the functions could be reduced ( possibly depending on the application ).
+        # the keypress waittime should probably be made shorter for these commands.
+        # When these commands are not working in a particular application sometimes the problem is that 
+        # there is not enough time from when control-c is pressed until the contents of the clipboard are passed into the function
+        # The solution is to add a longer pause after pressing control see in the supporting functions in text_manipulation_functions.py
+        # For some applications this pause ( and other pauses in the functions for that matter ) is not necessary
+        # and may be removed by the user if they wish to speed up the execution of these commands
+        
+
+
+# How to better consolidate these context actions so I'm not repeating myself so much???
+        
+        # PROBLEM: sometimes Dragon thinks the variables are part of dictation.      
+        "(replace|to) <lease_ross> [<number_of_lines_to_search>] <dictation> (with|to) <dictation2>":
+            R(ContextAction(default=Function(text_manipulation_functions.copypaste_replace_phrase_with_phrase,
+                       dict(dictation="replaced_phrase", dictation2="replacement_phrase", lease_ross="left_right"), 
+                       cursor_behavior="standard"), actions=[
+                        # Use different cursor method for texstudio
+                        (AppContext(executable="texstudio"), Function(text_manipulation_functions.copypaste_replace_phrase_with_phrase,
+                       dict(dictation="replaced_phrase", dictation2="replacement_phrase", lease_ross="left_right"), 
+                       cursor_behavior="texstudio"))]),
+              rdescript="Core: replace text to the left or right of the cursor"),
```

Comment:
This repetition is fine for now. Might be worth addressing if the number of contexts explodes.